examples/vision/image_enhancer.py

```diff
@@ -0,0 +1,276 @@
+"""
+Title: Document Images Restoration/Enhancer.
+Author: [Anish B](https://twitter.com/anishhacko)
+Date created: 2021/01/18
+Last modified: 2022/01/21
+Description: Example of Document Image Restoration coupled with Tesseract OCR engine to improve text readability.
+"""
+"""
+## Introduction
+
+This example explains a "Image Enhancer/Denoiser" using UNet architecture. The problem we
+focus here is enhancing the document images which are deteriorated through external
+degradations like blur, corrupt text blocks etc... ,we have followed up on few seminal
+papers as reference and used simple pretrained UNet as encoder and Efficient Sub-Pixel
+CNN as decoder. we do not attempt super-resolution as primary task hence we return output
+image in the same dimension as input. we compute WER (Word Error Rate) metric to evaluate
+model performance helping us to assess the text information gain in real world scenarios
+when coupled with tesseract OCR engine.
```

Comment:
@anish9 (cc. @fchollet )
The introduction lacks research information in order to be useful enough for the readers.  It would be nice to make a section (paragraph) for each: problem formulation, current research state, at the end of the notebook what reader will learn in general).